### ISSUE TITLE
build from source: uninstall 

### ISSUE THREAD

**Issue by aajjbb** [awesomeWM/awesome#1179]
Hi, I'm building awesome from source from the first time, and I realized that there's no 'uninstall' entry in the Makefile. 

After running `make install`, if I want to back and use awesome from my package manager, do I have to delete all the files by hand or there's a simpler way to do it ?

Thanks :)

**Comment by psychon** [comment]
CMake says "no": https://cmake.org/Wiki/CMake_FAQ#Can_I_do_.22make_uninstall.22_with_CMake.3F

If you can find the file mentioned there (in the source dir inside of `build/`, I guess), you can try `xargs rm < install_manifest.txt`.

By the way: There is `make package` which can produce .deb and .rpm packages from a source dir.

**Comment by aajjbb** [comment]
It does work, thank you. I'd say it's a little 'too much' for an uninstall. Shouldn't we list this link the README ?

**Comment by blueyed** [comment]
Since we have a cmake-wrapping Makefile we could just add it there, probably including some warning that it might be outdated and a pointer to better methods.

I personally use a PKGBUILD for Arch locally, and have some `awesome-rebuild` script to build it from the Git checkout.

@aajjbb 
What distro are you using?

**Comment by aajjbb** [comment]
@blueyed Manjaro :smiley: 
Maybe it could be interesting to keep an awesome-git package in **aur** ? I don't have numbers but I believe that Arch-bases distros users are a big part of awesome community.

**Comment by blueyed** [comment]
For Arch, there is https://aur.archlinux.org/packages/awesome-git/ already (but flagged out of date currently?!)
I became also the maintainer for https://aur.archlinux.org/packages/awesome-luajit-git/, when asking what the -git should apply to.  I could imagine updating and pushing that (to be awesome-git with luajit).

**Comment by aajjbb** [comment]
Oh, nice to hear it. Ironically, _awesome-git_ is behind the official stable package (is it true or only a wrong tag ?). I'll move to git as soon as it gets updated.

Thanks.

**Comment by psychon** [comment]
> Oh, nice to hear it. Ironically, awesome-git is behind the official stable package (is it true or only a wrong tag ?). I'll move to git as soon as it gets updated.

Not really. Looking at the PKGBUILD, it seems to build the latest git/master version at the time the package is build (does it really? I'm only 75% sure).
The version number might be due to "historic reasons". 3.5.2 was the last release before the master and 3.5 branches diverges, so it is the latest tag that "git describe" finds when looking at the master branch. Thus, it describes git master as "version 3.5.2 plus about 2000 changes".

I'm not sure about an "uninstall" target. It is better than nothing, but the CMake-FAQ seems to discourage it.

**Comment by aajjbb** [comment]
> I'm not sure about an "uninstall" target. It is better than nothing, but the CMake-FAQ seems to discourage it.

They elaborate that it might remove useful files from the system, but keeping files from a program I don't want isn't good either.

I decided to get awesome-git from aur, but now the build fails with an error I'm not sure about [here](http://pastie.org/10949862)

**Comment by psychon** [comment]
Can you try a different ldoc version (or perhaps just uninstall ldoc?)

**Comment by aajjbb** [comment]
Uninstall ldoc and remove it from PKGBUILD worked. 

But now `awesome --version` shows me: 

`awesome v3.5.2-2155-gf71592a (The Fox)

• Compiled against Lua 5.3.3 (running with Lua 5.3)
• D-Bus support: ✔
`

Is it just a tagging issue ?

**Comment by psychon** [comment]
Yup. The point were the master branch and the 3.5 branch diverged was some time after awesome 3.5.2. So, that tag is the last reachable tag from master (and you have 2155 commits since then, that's quite a number...). Sorry for this weird version.

**Comment by aajjbb** [comment]
Great, I'll stick with this version right now. I'm experiencing some memory issues using awesome 3.5.9, hope they are gone in git version.

**Comment by blueyed** [comment]
Cool.
Should we add`uninstall` to our Makefile, proxying to CMake, or leave it as-is, and just close this issue?
